sources/web/datalab/server.ts

```diff
@@ -141,6 +152,11 @@ function handleRequest(request: http.ServerRequest,
       (path.indexOf('/files') == 0) ||
       (path.indexOf('/edit') == 0) ||
       (path.indexOf('/sessions') == 0)) {
+
+    if (path.indexOf('/tree') == 0) {
+        loadedSettings[startup_path_setting] = path
+        settings_.updateUserSetting(userId, startup_path_setting, path);
```

Comment:
There's no reliable way to execute code before killing the process, as far as I know, so I'll just turn this into an async call.